src/Bindings/LuaJson.cpp

```diff
@@ -0,0 +1,321 @@
+
+// LuaJson.cpp
+
+// Implements the Json exposure bindings to Lua
+
+#include "Globals.h"
+#include <sstream>
+#include "LuaJson.h"
+#include "LuaState.h"
+#include "tolua++/include/tolua++.h"
+#include "json/json.h"
+
+
+
+
+
+// fwd:
+
+static void PushJsonValue(const Json::Value & a_Value, cLuaState & a_LuaState);
+static Json::Value JsonSerializeValue(cLuaState & a_LuaState);
+
+
+
+
+
+/** Pushes the specified Json array as a table on top of the specified Lua state.
+Assumes that a_Value is an array. */
+static void PushJsonArray(const Json::Value & a_Value, cLuaState & a_LuaState)
+{
+	// Create the appropriately-sized Lua table:
+	lua_createtable(a_LuaState, static_cast<int>(a_Value.size()), 0);
+
+	// Insert each value to the appropriate index (1-based):
+	int idx = 1;
+	for (const auto & v: a_Value)
+	{
+		if (v.isNull())
+		{
+			// Skip null values
+			continue;
+		}
+
+		PushJsonValue(v, a_LuaState);
+		lua_rawseti(a_LuaState, -2, idx);
+		idx += 1;
+	}  // for v: a_Value[]
+}
+
+
+
+
+
+/** Pushes the specified Json object as a table on top of the specified Lua state.
+Assumes that a_Value is an object. */
+static void PushJsonObject(const Json::Value & a_Value, cLuaState & a_LuaState)
+{
+	// Create the appropriately-sized Lua table:
+	lua_createtable(a_LuaState, 0, static_cast<int>(a_Value.size()));
+
+	// Json::Value has no means of iterating over children with their names included.
+	// We need to iterate over names and "find" them in the object again:
+	auto names = a_Value.getMemberNames();
+	for (const auto & n: names)
+	{
+		auto v = a_Value[n];
+		if (v.isNull())
+		{
+			// Skip null values
+			continue;
+		}
+
+		// Set the value in Lua's table:
+		a_LuaState.Push(n);
+		PushJsonValue(v, a_LuaState);
+		lua_rawset(a_LuaState, -3);
+	}  // for itr - a_Value[]
+}
+
+
+
+
+
+/** Pushes the specified Json value as an appropriate type on top of the specified Lua state.
+Assumes that the Json value is not a null. */
+void PushJsonValue(const Json::Value & a_Value, cLuaState & a_LuaState)
+{
+	switch (a_Value.type())
+	{
+		case Json::nullValue:
+		{
+			ASSERT(!"Cannot push null json value onto Lua stack");
```

Comment:
I decided to convert `Json::nullValue` to a Lua `nil`, it doesn't seem that would break anything, other than arrays having holes in them, which is acceptable I guess.